src/optimistic-oracle/optimistic-oracle.js

```diff
@@ -0,0 +1,232 @@
+        findings.push(Finding.fromObject({
+          name: 'UMA Price Request',
+          description: `Price requested from Optimistic Oracle.  Currency=${currency}, Price=${price}`,
+          alertId: 'AE-UMA-OO-REQUESTPRICE',
+          severity: FindingSeverity.Low,
+          type: FindingType.Unknown,
+          protocol: 'uma',
+          everestId: config.umaEverestId,
+          metadata: {
+            requester,
+            currency,
+            price,
+          },
+        }));
+      }
+
+      if (log.name === 'ProposePrice') {
+        const { currency } = log.args;
+        const { requester } = log.args;
+        const { proposer } = log.args;
+
+        // set up ERC-20 contract to get decimals value
+        // in production, erc20Contract will always be undefined
+        // in testing, erc20Contract will be assigned to a mock contract
+        /* eslint-disable no-param-reassign */
+        if (erc20Contract === undefined) {
+          erc20Contract = new ethers.Contract(currency, erc20Abi, provider);
+        }
+        /* eslint-enable no-param-reassign */
+
+        const decimals = await erc20Contract.decimals();
+
+        // convert proposedPrice to a human-readable decimal value
+        let proposedPrice = ethers.utils.formatUnits(log.args.proposedPrice, decimals);
+
+        // convert the proposedPrice to a BigNumber type for difference calculations later
+        proposedPrice = new BigNumber(proposedPrice.toString());
+
+        // make a request to the price feed API
+        // CoinGecko will return decimal values
+        let price;
+        try {
+          price = await getPrice(currency);
+        } catch (error) {
+          console.error(error);
+          continue;
+        }
+
+        // convert the price obtained to a BigNumber type
+        price = new BigNumber(price);
+
+        // generate a finding if the price difference threshold (defined in agent-config.json) has
+        // been exceeded
+        const percentError = calculatePercentError(proposedPrice, price);
+
+        if (percentError.isGreaterThan(config.optimisticOracle.priceThresholdPct)) {
+          findings.push(Finding.fromObject({
+            name: 'UMA Price Proposal',
+            description: `Price proposed to Optimistic Oracle is disputable. Currency=${currency}, ProposedPrice=${proposedPrice}, Price=${price}`,
+            alertId: 'AE-UMA-OO-PROPOSEPRICE',
+            severity: FindingSeverity.Low,
+            type: FindingType.Unknown,
+            protocol: 'uma',
+            everestId: config.umaEverestId,
+            metadata: {
+              requester,
+              proposer,
+              currency,
+              proposedPrice: proposedPrice.toString(),
+              price: price.toString(),
+            },
+          }));
+        }
+      }
+    }
+
+    return findings;
+  };
+}
+
+module.exports = {
+  provideHandleTransaction,
+  handleTransaction: provideHandleTransaction(undefined),
```

Comment:
No change needed here (I prefer explicit over implicit), but I thought I would mention some alternatives that I came upon:
Javascript supports default argument values: `provideHandlerTransaction(erc20Contract=undefined)` would work
Function parameters default to `undefined` if not specified.  Calling `provideHandlerTransaction()` (no argument) would result in `erc20Contract` being `undefined`

Reply:
big fan of just not passing in anything and not putting a default in the `provideHandlerTransaction` signature

Reply:
Changed